lib/utils.js

```diff
@@ -2,6 +2,22 @@
 const MongoError = require('./core/error').MongoError;
 const ReadPreference = require('./core/topologies/read_preference');
 const WriteConcern = require('./write_concern');
+const maxWireVersion = require('./core/utils').maxWireVersion;
+
+const serverLacksFeature = function(server, feature) {
```

Comment:
Not sure how others feel about this helper, but I think it cleans things up - it's annoying having version checks for features scattered throughout the source, written in all sorts of different ways that makes it difficult even grepping for them. Centralizing them all at once would probably be a large effort, but a helper like this may help us achieve it more progressively.

Reply:
Perhaps I should be using `ServerCapabilities` then? I'm not a fan of how often there is a `maxWireVersion` check against a straight up number in the code, pretty agnostic about how it gets centralized; though I do think it's nice to have the error strings in one place too, when they are generic at least.

Reply:
Perhaps within `ServerCapabilities`, in addition to the "has" prefix a "supports" prefix? `hasCollation` will be a boolean, and supportsCollation a method that throws an error?

Reply:
I'm not sure throwing makes sense, but we could make the prefix `lacks` and it could either return an Error or undefined.

I just want to be sure `ServerCapabilities` is 100% equivalent to doing a `maxWireVersion(server)` or `maxWireVersion(topology)`, which is what we generally do in these cases as far as I've seen. So that I can confidently replace code of the pattern:

```
    if (maxWireVersion(any) < 9) {
      callback(new MongoError('MONGODB-AWS authentication requires MongoDB version 4.4 or later'));
      return;
    }
```

with something like:
```
const awsAuthNotSupportedErr = capabilities(connection).lacksAwsAuth;
if (awsAuthNotSupportedErr) {
  callback(awsAuthNotSupportedErr);
  return;
}
```

Reply:
I think my greater point was that this change feels a bit out of scope for this ticket. I agree we need a better solution, but if we're adding this in here without fully thinking it out then we just have 3 solutions which aren't exactly what we want where we used to have 2. I'd prefer to stick with the direct check in this PR, and come up with a solution we're all really happy with and outright remove `ServerCapabilities`, `checkForCollation`, etc etc